### PR TITLE
SD-4847: Update monitoring items when an item is sent to another stage/desk.

### PR DESCRIPTION
**Pull request by mscam**
* Update monitoring items when an item is sent to another stage/desk.

### CODE REVIEW

File: scripts/superdesk-desks/desks.js [hunk 1]
```diff
@@ -216,15 +216,18 @@
                  *
                  * In case it gets called multiple times it will query only once
                  */
-                function scheduleQuery() {
+                function scheduleQuery(delay) {
```
Comment (gbbr) [inline scripts/superdesk-desks/desks.js]: You can do:
```js
function scheduleQuery(delay = 5000) {
```
to obtain default parameter values, and remove the `if` statement below.

File: scripts/superdesk-desks/desks.js [hunk 2]
```diff
@@ -172,7 +172,7 @@
                 }
 
                 scope.$watch('filter', queryItems);
-                scope.$on('task:stage', function(event, data) {
+                scope.$on('task:stage', function($event, data) {
```
Comment (gbbr) [inline scripts/superdesk-desks/desks.js]: Why this change? The parameter is not used.

**Comment by gbbr** [comment]
Can you please squash all the commits into one?

**Comment by gbbr** [comment]
Many times broadcasting on `$rootScope` is not necessary - do you think this might be one of those cases? I think the server already sends `item:move` via WebSockets, which is automatically broadcasted already. Maybe it's not picked up?

screen shot 2016-08-18 at 17 58 50

File: scripts/superdesk-desks/desks.js (hunk 2 above)
Comment (mscam) [inline scripts/superdesk-desks/desks.js]: Consistency with the other event listeners that are using $event as argument name, no particular reason.

**Comment by mscam** [comment]
I can squash commits with github with the merge button, but I can also rebase my feature branches in future.

**Comment by mscam** [comment]
I am not against broadcasting events to $rootScope, sometimes is useful. But it's also good to discover why that event is not picked up. What do you think? Let's discuss about the WebSockets  issue tomorrow.

**Comment by gbbr** [comment]
Sure, let's do a hangout

File: scripts/superdesk-desks/desks.js (hunk 2 above)
Comment (gbbr) [inline scripts/superdesk-desks/desks.js]: The event listeners that you see using `$event` are using Angular's [`$event`](https://docs.angularjs.org/guide/expression#-event-), which is fine. In this case this is a custom event so `e` or `event` should be better to avoid confusing the two concepts. Just like above where you've used `e` yourself.

File: scripts/superdesk-archive/module.js [hunk 3]
```diff
@@ -91,8 +91,9 @@ function MultiService($rootScope) {
     $rootScope.$on('item:publish', function(e, data) {
         self.remove(data.item);
     });
-
-    //ToDo: Track upcoming item:move event as well for updateCount.
```
Comment (gbbr) [inline scripts/superdesk-archive/module.js]: 👍 nice one

File: scripts/superdesk-desks/desks.js (hunk 2 above)
Comment (mscam) [inline scripts/superdesk-desks/desks.js]: ok great, I will change all the relevant unused argument name respecting that naming convention.

**Comment by gbbr** [comment]
Much better, thank you @mscam - now if we could make it all into one commit, then we can merge. I assume this will give many conflicts when attempting to squash so it might be best to just create a new branch and copy over the changes to get a clean commit.

File: scripts/superdesk-desks/desks.js [hunk 4]
```diff
@@ -172,22 +172,22 @@
...
                 });
 
-                scope.$on('item:move', function($event, data) {
-                    if ((data.to_stage && data.to_stage === scope.stage) ||
-                        (data.from_stage && data.from_stage === scope.stage)) {
-                        scheduleQuery();
+                scope.$on('item:move', function(_e, data) {
+                    if ((data.to_desk && data.from_desk !== data.to_desk) ||
+                        (data.to_stage && data.from_stage !== data.to_stage))  {
+                        scheduleQuery(1000); // smaller delay.
```
Comment (petrjasek) [inline scripts/superdesk-desks/desks.js]: it should be a bit more, like 2s, it takes some time to elastic after a change to show the updates

**Comment by gbbr** [comment]
Closing in favour of https://github.com/superdesk/superdesk-client-core/pull/870

File: scripts/superdesk-desks/desks.js (hunk 4 above)
Comment (mscam) [inline scripts/superdesk-desks/desks.js]: Ok, thanks.
